### PR TITLE
add camera pan method

### LINKED ISSUE #1430: Add a Pan method
**Is your feature request related to a problem? Please describe.**
When having an external window there is not method to pan.

**Describe the solution you'd like**
A Pan method where you pass the delta x and y to pan vertically or horizontally.

**Describe alternatives you've considered**
Calculate the new camera position and focal point in the external program.

**Additional context**
[Here](https://github.com/Nokse22/Exhibit/blob/3f6b2118d0365da4eeab1e8cc87d760cb5c54dbd/src/window.py#L204-L212) is the solution thanks to @snoyer that I have in my app.

### PR DESCRIPTION
**Pull request by snoyer**
fixes #1430

### CODE REVIEW

File: library/public/camera.h [hunk 1]
```diff
@@ -57,6 +57,7 @@ class F3D_EXPORT camera
    * Angles are in degrees.
    */
   virtual camera& dolly(double val) = 0;
+  virtual camera& pan(double right, double up) = 0;
```
Comment (mwestphal) [inline library/public/camera.h:60]: I wonder if dolly and pan should be separately documented to clarify the arguments

File: python/testing/test_camera.py [hunk 2]
```diff
@@ -86,6 +86,21 @@ def test_moves():
     camera.pitch(angle)
 
 
+def test_pan():
```
Comment (mwestphal) [inline python/testing/test_camera.py:89]: nice, but needs a C++ test :)

**Review by mwestphal — CHANGES_REQUESTED** [review]
small changes needed

File: library/public/camera.h (hunk 1 above)
Comment (snoyer) [reply library/public/camera.h:60]: would changing `dolly(double val)` to `dolly(double forward)` be self-explanatory enough?

File: python/testing/test_camera.py (hunk 2 above)
Comment (snoyer) [reply python/testing/test_camera.py:89]: for some reason I thought python testing counted towards coverage

File: library/public/camera.h (hunk 1 above)
Comment (mwestphal) [reply library/public/camera.h:60]: definitely an improvement.

File: python/testing/test_camera.py (hunk 2 above)
Comment (mwestphal) [reply python/testing/test_camera.py:89]: The logic is that python is an optional dependency, so python tests are testing the bindings themselves, not the feature, which must be tested using c++

File: library/public/camera.h (hunk 1 above)
Comment (mwestphal) [reply library/public/camera.h:60]: unrelated to that PR though

File: library/testing/TestSDKCamera.cxx [hunk 3]
```diff
@@ -27,6 +28,31 @@ bool comparePoint(const f3d::point3_t& vec1, const f3d::point3_t& vec2)
...
+  if (!compareDouble(actual[0], expected[0]) || !compareDouble(actual[1], expected[1]) ||
+    !compareDouble(actual[2], expected[2]))
+  {
+    std::stringstream ss;
+    ss << label << ": ";
+    ss << std::setprecision(12);
+    ss << "(" << actual[0] << "," << actual[1] << "," << actual[2] << ")";
+    ss << " != ";
+    ss << "(" << expected[0] << "," << expected[1] << "," << expected[2] << ")";
+    throw testFailure(ss.str());
```
Comment (mwestphal) [inline library/testing/TestSDKCamera.cxx:52]: any specific reason to go the exceptions instead of conditions (apart from your personnal preference) ?

Comment (snoyer) [reply library/testing/TestSDKCamera.cxx:52]: feels like it's less noise to throw on failure and have a single `try {...} catch { std::cerr<< what(); return EXIT_FAILURE; } return EXIT_SUCCESS;` than to have a bunch of `if(!...){ std::cerr<< ... ; return EXIT_FAILURE; }`
...so personal preference mostly I guess

**Review by mwestphal — APPROVED** [review]

**Review by Meakk — APPROVED** [review]

File: library/public/camera.h (hunk 1 above)
Comment (snoyer) [inline library/public/camera.h:60]: I'll add a forward parameter for completeness
```suggestion
  virtual camera& pan(double right, double up, double forward = 0) = 0;
```

**Review by mwestphal — APPROVED** [review]

**Review by Meakk — APPROVED** [review]

**Pull request merged**